pkg/test/integration/common/framework.go

```diff
@@ -0,0 +1,815 @@
+		if targetKubeConfigPath == "" {
+			targetKubeConfigPath = controlKubeConfigPath
+			log.Println("Missing targetKubeConfig. control cluster will be set as target too")
+		}
+		targetKubeConfigPath, _ = filepath.Abs(targetKubeConfigPath)
+		// use the current context in controlkubeconfig
+		var err error
+		c.ControlCluster, err = helpers.NewCluster(controlKubeConfigPath)
+		if err != nil {
+			return err
+		}
+		c.TargetCluster, err = helpers.NewCluster(targetKubeConfigPath)
+		if err != nil {
+			return err
+		}
+
+		// update clientset and check whether the cluster is accessible
+		err = c.ControlCluster.FillClientSets()
+		if err != nil {
+			log.Println("Failed to check nodes in the cluster")
+			return err
+		}
+
+		err = c.TargetCluster.FillClientSets()
+		if err != nil {
+			log.Println("Failed to check nodes in the cluster")
+			return err
+		}
+	}
+
+	// Update namespace to use
+	if c.ControlCluster.IsSeed(c.TargetCluster) {
+		_, err := c.TargetCluster.ClusterName()
+		if err != nil {
+			log.Println("Failed to determine shoot cluster namespace")
+			return err
+		}
+		controlClusterNamespace, _ = c.TargetCluster.ClusterName()
+	} else if controlClusterNamespace == "" {
+		controlClusterNamespace = "default"
+	}
+	return nil
+}
+
+func (c *IntegrationTestFramework) prepareMcmDeployment(mcContainerImageTag string, mcmContainerImageTag string, byCreating bool) error {
+	/*
+		if any of mcmContainerImage  or mcContainerImageTag flag is non-empty then,
+		update machinecontrollermanager deployment in the control-cluster with specified image
+		 -
+	*/
+	if byCreating {
+		// Create clusterroles and clusterrolebindings for control and target cluster
+		// Create secret containing target kubeconfig file
+		// Create machine-deployment using the yaml file
+		c.ControlCluster.ControlClusterRolesAndRoleBindingSetup(controlClusterNamespace)
+		c.TargetCluster.TargetClusterRolesAndRoleBindingSetup()
+		configFile, _ := os.ReadFile(c.TargetCluster.KubeConfigFilePath)
+		c.ControlCluster.Clientset.CoreV1().Secrets(controlClusterNamespace).Create(&coreV1.Secret{
+			ObjectMeta: metav1.ObjectMeta{
+				Name: "machine-controller-manager-target",
+			},
+			Data: map[string][]byte{
+				"kubeconfig": configFile,
+			},
+			Type: coreV1.SecretTypeOpaque,
+		})
+
+		err := c.ControlCluster.
+			ApplyFile("../../../kubernetes/deployment.yaml",
+				controlClusterNamespace)
+		if err != nil {
+			return err
+		}
+	}
+
+	// mcmDeploymentOrigObj holds a copy of original mcm deployment
+	result, getErr := c.ControlCluster.Clientset.AppsV1().Deployments(controlClusterNamespace).Get("machine-controller-manager", metav1.GetOptions{})
+	if getErr != nil {
+		log.Printf("failed to get latest version of Deployment: %v", getErr)
+		return getErr
+	}
+	mcmDeploymentOrigObj = result
+
+	// update containers spec
+	providerSpecificRegexp, _ := regexp.Compile("machine-controller-manager-provider-")
```

Comment:
This is not a good assumption as container names as different in all [providers](https://github.com/gardener/gardener-extension-provider-openstack/blob/master/charts/internal/machine-controller-manager/seed/templates/deployment.yaml#L84). If it's a small change can you please create an ENV variable called `MC_CONTAINER_NAME`? And the default value you can keep it to `machine-controller-manager-provider-aws`? THis way you could also avoid the regexp-based checking also.

Reply:
The regex match is tried against container image [here](https://github.com/gardener/machine-controller-manager/pull/618/files#). So this assumption is for container image and not its name.
Is this ok?